docs/overview/timeline.md

```diff
@@ -467,6 +467,104 @@ const after_if_trial = {
 jsPsych.run([pre_if_trial, if_node, after_if_trial]);
 ```
 
+## Modifying timelines at runtime
+
+Although this functionality can also be achieved through a combination of the `conditional_function` and the use of dynamic variables in the `stimulus` parameter, our timeline implementation allows you to dynamically insert or remove trials and nested timelines during runtime.
+
+### Inserting timeline nodes at runtime
+For example, you may have a branching point in your experiment where the participant is given 3 choices, each leading to a different timeline: 
+
+```javascript
+const jspsych = initJsPsych();
+let timeline = [];
+
+const welcome_trial = {
+	type: jsPsychHtmlKeyboardResponse,
+	stimulus: 'Welcome!'
+}
+
+const choice_trial = {
+	type: jsPsychHtmlKeyboardResponse,
+	stimulus: 'Press 1 if you are a new participant. Press 2 for inquiries about an existing experiment run. Press 3 for Spanish.',
+	choices: ['1','2','3']
+}
+```
+This would be trickier to implement with the `conditional_function` since it can only handle 2 branches -- case when `True` or case when `False`. Instead, you can modify the timeline by modifying `choice_trial` to dynamically adding a timeline at the end of the choice trial according to the chosen condition:
+
+```javascript
+const b1_t1 = {...};
+const b1_t2 = {...};
+const b1_t3 = {...};
+// So on and so forth
+const b3_t3 = {...};
+
+const branch_1 = [b1_t1, b1_t2, b1_t3];
+const branch_2 = [b2_t1, b2_t2, b2_t3];
+const branch_3 = [b3_t1, b3_t2, b3_t3];
+
+const choice_trial = {
+	type: jsPsychHtmlKeyboardResponse,
+	stimulus: 'Press 1 for English. Press 2 for Mandarin. Press 3 for Spanish.',
+	choices: ['1','2','3'],
+	on_finish: (data) => {
+		switch(data.response) {
+			case '1':
+				timeline.push(branch_1);
+				break;
+			case '2':
+				timeline.push(branch_2);
+				break;
+			case '3':
+				timeline.push(branch_3);
+				break;
+		}
+	}
+}
+timeline.push(start_trial, choice_trial);
+```
+During runtime, choices 1, 2 and 3 will dynamically add a different (nested) timeline, `branch_1`, `branch_2` and `branch_3` respectively, to the end of the main timeline.
+
+### Removing timeline nodes at runtime
+
+You can also remove upcoming timeline nodes from the timeline at runtime. To demonstrate this, we can modify the above example by adding a 4th choice and extra trial at the end of the timeline:
+
+```javascript
+const choice_trial = {
+	type: jsPsychHtmlKeyboardResponse,
+	stimulus: 'Press 1 for English. Press 2 for Mandarin. Press 3 for Spanish. Press 4 to exit.',
```

Comment:
when would exiting make sense?

and wouldn't the pop remove the end of experiment trial? that seems like exiting without a confirmation or something? 

I guess I'm having a hard time thinking of when removal would be something that is worth doing. We could document it as: we don't really think there's a use case for this, but just in case you come up with one, you can also remove a trial... canned example...

Reply:
yea pop removes the end of experiment trial, which was added to the tail of the timeline in the first place to demonstrate the removal. maybe if i switch it to something like "part 2" it would feel more natural?